odoo/addons/base/models/res_currency.py

```diff
@@ -272,3 +280,27 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
                 name = ''
                 operator = 'ilike'
         return super(CurrencyRate, self)._name_search(name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid)
+        
+    @api.depends('currency_id.rate_ids')
+    def _compute_date_end(self):
+        currency_ids = self.mapped('currency_id').ids
+        
+        self.env.cr.execute("""
```

Comment:
Making this query in SQL would require any kind on cache invalidation, isn't it? Isn't better to use ORM for the write (and maybe `read_group` for the read)?